### PR TITLE
Validation for price field in social event dialog

### LINKED ISSUE #1863: Validation for price missing in social event dialog
While editing the registration form a social event can be configured as billable and empty string as price. This causes an exception on form submission that prevents users from registering. We should add validation in the client-side and also default to 0 in any case when received.

### PR DESCRIPTION
**Pull request by mic4ael**
fixes #1863

### CODE REVIEW

File: indico/htdocs/css/Default.css [hunk 1]
```diff
@@ -6783,3 +6783,7 @@ ul.speaker_list > li .contributions li {
     width: 400px;
     height: 200px;
 }
+
+input[name="price"].ng-invalid-required {
```
Comment (OmeGak) [inline indico/htdocs/css/Default.css]: I wouldn't add new css rules to `Default.css`. I think this is the place to put it: `htdocs/sass/modules/_registrationform.scss`.

File: indico/htdocs/js/indico/RegistrationForm/section.js [hunk 2]
```diff
@@ -797,7 +797,13 @@ ndRegForm.directive('ndSectionDialog', function(url) {
             };
 
             $scope.addItem = function () {
-                 $scope.formData.items.push({id:'isNew', cancelled: false});
+                 $scope.formData.items.push({
+                    id: 'isNew',
+                    cancelled: false,
+                    price: '0',
```
Comment (OmeGak) [inline indico/htdocs/js/indico/RegistrationForm/section.js]: This looks wrong, why would you add `price` to all section dialogs?

File: indico/MaKaC/registration.py [hunk 3]
```diff
@@ -4929,7 +4929,7 @@ class BilledItem(object):
 
     def __init__(self, caption, price, quantity, currency):
```
Comment (OmeGak) [inline indico/MaKaC/registration.py]: The correct way to assign default price would be:

```python
def __init__(self, caption, quantity, currency, price=0.0):
```

Just make sure to update the order of arguments passed to `BilledItem()`.

File: indico/htdocs/js/indico/RegistrationForm/section.js (hunk 2 above)
Comment (OmeGak) [inline indico/htdocs/js/indico/RegistrationForm/section.js]: We could do something like this to add arbitrary entries to the dict:

```js
$scope.addItem = function(args) {
    $scope.formData.items.push($.extend({id:'isNew', cancelled: false}, args));
};
```

Comment (mic4ael) [inline indico/htdocs/js/indico/RegistrationForm/section.js]: Might be useful ;)

**Comment by pferreir** [comment]
Merged in 090d7b3dc562bfa7f5abf5f69a89088f159d8014.

**Pull request closed without merging**